src-v2/Judge/NTS.Judge.Blazor/Pages/Dashboard/Components/Actions/EliminationForms/DisqualifyForm.razor

```diff
@@ -14,32 +15,18 @@
 </MudButton>
 
 @code {
-
-    [Inject]
-    IParticipationBehind _participationBehind { get; set; } = default!;
     [Parameter]
     public Disqualified? Disqualified { get; set; }
-    [Parameter]
-    public EventCallback OnRestore { get; set; }
-    [Parameter]
-    public EventCallback OnEliminate { get; set; }
     private string? Reason { get; set; }
 
     protected override void OnParametersSet()
     {
         Reason = Disqualified?.Complement;
     }
 
-    public async Task Eliminate()
+    public override async Task Eliminate()
     {
-        Participation? participation = _participationBehind.SelectedParticipation;
-        var tandemNumber = participation!.Tandem.Number;
-        await _participationBehind.Disqualify(tandemNumber, Reason);
+        await _participationBehind.Disqualify(Reason);
```

Comment:
What if reason is null here?

Reply:
This should be validated. Adding To Dos for validation. An abstraction of some sort for NotForm<T> is probably needed to enable the use of RegisterValidationInjectors for the EliminationForms as well.